assets/scripts/whichx.js

```diff
@@ -115,27 +147,38 @@ function WhichX(config) {
             }
             return bestLabel;
         } else {
-            throw new Error("Invalid description");
+            throw new Error("Invalid description " + description + " of type " + typeof description + ". We expected a non empty string.");
         }
     };
 
-    // Exports the WhichX internal data representation learned from provided
-    // labeled text. Please see the typesMap comments for more details.
+    /**
+     * Exports the WhichX internal data representation learned from provided
+     * labeled text. Please see the typesMap comments for more details.
+     * @returns {TypeMap} A TypeMap that can be sringified and saved for later import in WhichX
+     */
     this.export = function() {
         return typesMap;
     };
 
-    // Imports a previously exported typesMap. This will write over any data this instance has already learned.
+    /**
+     * Imports a previously exported typesMap. This will write over any data this instance has already learned.
+     * @param {TypeMap} importedTypesMap The types map previously exported from WhichX
+     */
     this.import = function(importedTypesMap) {
         var newTotal = importedTypesMap.total;
-        if (newTotal === "undefined" || newTotal.tcount === "undefined" || newTotal.wordTotal === "undefined") {
+        if (newTotal === undefined || newTotal.tcount === undefined || newTotal.wordTotal === undefined) {
             throw new Error("Import invalid. This doesn't look like it was exported from a prior model.");
         }
         typesMap = importedTypesMap;
     };
 
-    // Loop through words and work out probability of type given each word.
-    // Multiply each word's probability by total probability to determine type probability.
+    /**
```

Comment:
`getTypeChance()` is not available to call on the `WhichX` object, so the JSDoc here may not be all the helpful. Can we remove the docs to keep file size down?

Reply:
This also document the project for potential contributors, and can detect mistakes through type checking in PRs. If size is a concern, I advise using a minifier and a build process to the project.